Cockpit 265 (#504)
cockpit-project/cockpit-project.github.io: _posts/2022-03-16-cockpit-265.md

Support for this feature must be explicitly enabled via the new `ForwardedForHeader` option in the [configuration file](https://cockpit-project.org/guide/latest/cockpit.conf.5.html).

## Manifest overrides in /etc and ~/.config directories

[Manifest override files](https://cockpit-project.org/guide/latest/packages.html#package-replace) are a way to customize or hide Cockpit's page menu entries. Until now these needed to be put next to the `manifest.json` which they modify, for example `/usr/share/cockpit/systemd/override.json`. However, `/usr/share/` is package manager territory,  and admins should not (and in some cases even cannot) write files there. With this Cockpit version, these overrides can now be placed into `/etc/cockpit/systemd.override.json` for global customizations, or `~/.config/cockpit/systemd.override.json` for user specific ones.

## Crypto policies support

Cockpit now has basic support for displaying and changing a subset of crypto policies (default, future, legacy). The current configured crypto policy is shown in the "Configuration" card.

> ![158211328-fd8936be-5127-4e30-9c4b-5ba251c7d60a]({{ site.baseurl }}/images/265-crypto-policies-support.png)

Reviewer: This image is blurry, and it couldn't hurt to equalize the frame width a bit.

Also, please let's not use these UUIDs as alt names, but something readable like "crypto policies in Configuration card"

Reviewer: Please still fix these horrible UUIDs.

Crypto policy changes can be applied immediately or later. When "Apply only" is selected, the Health card shows that a reboot is required to fully apply the changed crypto policy.

> ![158211149-36822bc4-a47e-4501-a01a-a82f7bbf3ec2]({{ site.baseurl }}/images/265-crypto-policies-support-2.png)

Reviewer: This is blurry as well, looks like being scaled up? Same UUID alt-name issue.

---
title: Cockpit 265
author: skobyda
date: '2022-03-16'
tags: cockpit, machines
slug: cockpit-265
category: release

> summary: 'Animated lists, x-forwarder-for support, crypto policies, disk serial number, metrics showing busiest CPU core, manifest overrides location'

Reviewer: Seems a little longish, but then again we don't actually show these summaries anywhere. But it's X-Forwarded-For, not lower-case and forwarde*r*

---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 265 and cockpit-machines 264:

## Animate new rows in lists

When new items appear on a page, they are highlighted for a short while to spot them more easily.

![Peek 2022-03-15 07-47]({{ site.baseurl }}/images/265-animate-new-rows-in-lists.gif)

## Support for `X-Forwarded-For`

When run in a configuration involving a reverse proxy server, Cockpit can now report the client IP address as provided to it by the proxy server, rather than using the IP of the proxy server itself.  This impacts what appears in system login logs, for example, and may have an impact on authentication, depending on which pam modules are in use.

> Support for this feature must be explicitly enabled via the new `ForwardedForHeader` option in the configuration file.

Reviewer: Let's turn "configuration file" into a link to https://cockpit-project.org/guide/latest/cockpit.conf.5.html

## Manifest overrides in /etc and ~/.config directories

[Manifest override files](https://cockpit-project.org/guide/latest/packages.html#package-replace) are a way to customize or hide Cockpit's page menu entries. Until now these needed to be put next to the `manifest.json` which they modify, for example `/usr/share/cockpit/systemd/override.json`. However, `/usr/share/` is package manager territory,  and admins should not (and in some cases even cannot) write files there. With this Cockpit version, these overrides can now be placed into `/etc/cockpit/systemd.override.json` for global customizations, or `~/.config/cockpit/systemd.override.json` for user specific ones.

## Crypto policies support

Cockpit now has basic support for displaying and changing a subset of crypto policies (default, future, legacy). The current configured crypto policy is shown in the "Configuration" card.

![158211328-fd8936be-5127-4e30-9c4b-5ba251c7d60a]({{ site.baseurl }}/images/265-crypto-policies-support.png)

> Crypto policy changes can be applied directly or later, when later is selected the health card shows that a reboot is required to fully apply the changed crypto policy.

Reviewer: ```suggestion
Crypto policy changes can be applied immediately or later. When "Apply only" is selected, the Health card shows that a reboot is required to fully apply the changed crypto policy.
```

![158211149-36822bc4-a47e-4501-a01a-a82f7bbf3ec2]({{ site.baseurl }}/images/265-crypto-policies-support-2.png)

## Metrics: Show busiest CPU core

On multi-core systems, the Metrics page now shows an additional usage bar for the CPU core with the highest usage. This points out bottlenecks and performance problems which happen because only a few cores (or even a single one) get fully used, while many others are mostly idle. In that case, the average CPU usage will be low.

![screenshot of show busiest cpu core]({{ site.baseurl }}/images/265-metrics-show-busiest-cpu-core.png)

## Machines: Disk serial number

> Disks identifier can be specified during a disk attachment. Its use cases include:

Reviewer: ```suggestion
A disk identifier can be specified during attachment. Use cases include:
```

- User might have licensed functionality inside the VM tied to a specific
disk serial number (most common with Windows guests)
- Setting up multipath storage

> - This disk will be identifiable from inside the guest by /dev/disk/by-id.

Reviewer: ```suggestion
- This disk will be identifiable from inside the guest in `/dev/disk/by-id/`.
```

![TestMachinesDisks-testAddDiskAdditionalOptions-vm-add-disk-modal-additional-options-pixels]({{ site.baseurl }}/images/265-machines-disk-serial-number.png)

## Try it out

Cockpit 265 and cockpit-machines 264 are available now:

* [For your Linux system](https://cockpit-project.org/running.html)

* [Cockpit Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/265)
* [Cockpit Fedora 36](https://bodhi.fedoraproject.org/updates/FEDORA-2022-ad93e6cdcb)
* [Cockpit Fedora 35](https://bodhi.fedoraproject.org/updates/FEDORA-2022-ec9612ac99)
* [Cockpit Client](https://flathub.org/apps/details/org.cockpit_project.CockpitClient)
* [cockpit-machines Source Tarball](https://github.com/cockpit-project/cockpit-machines/releases/tag/264)
* [cockpit-machines Fedora 36](https://bodhi.fedoraproject.org/updates/FEDORA-2022-7e1cbde0f7)
* [cockpit-machines Fedora 35](https://bodhi.fedoraproject.org/updates/FEDORA-2022-2cd2b4d57d)

> * [Cockpit Client](https://flathub.org/apps/details/org.cockpit_project.CockpitClient)

Reviewer: We didn't actually release a new Client, and none of the new features affect it -- i.e. we don't need to. Let's just drop the line.
